packages/react/src/file-upload/FileUpload.tsx

```diff
@@ -282,22 +348,63 @@ export const FileUpload = forwardRef<HTMLInputElement, FileUploadProps>(
 									Select {filesPlural}
 								</Button>
 							</Flex>
-							{showFileLists && (
+							{areFileListsVisible && (
 								<Stack gap={0.5}>
-									<Text color="muted">{fileSummaryText}</Text>
+									{Boolean(allRejections.length) && (
+										<SectionAlert
+											title="The following files could not be selected"
+											tone="error"
+											onClose={clearErrors}
+										>
+											<Text as="p">
+												These files did not meet upload requirements, were not
+												listed, and will not be uploaded.
+											</Text>
+											<UnorderedList>
+												{allRejections.map((file) => {
+													return (
+														<ListItem key={file.file.name}>
+															<Text as="strong" color="error" fontWeight="bold">
+																{file.file.name}
+															</Text>{' '}
+															<span>
+																({formatFileSize(file.file.size)}){' - '}
+																{getErrorSummary(
+																	file.errors,
+																	formattedMaxFileSize
+																)}
+															</span>
+														</ListItem>
+													);
+												})}
+											</UnorderedList>
+										</SectionAlert>
+									)}
+									<Flex justifyContent="space-between" alignItems="center">
+										<Text color="muted">{fileSummaryText}</Text>
+										{/* This "remove all files" button requires more thought and discussion before implementing */}
```

Comment:
I think we should remove the commented out code. Don't throw it away, just put it in a follow up ticket where we discuss the implications of the feature.